### PR TITLE
ArcaneBungee v3.0

### LINKED ISSUE #9: Arcane v3.0 - ArcaneBungee Project
Commands to add:

* [x] /apply
* [x] /badge (/b, /tag)
* [x] /badgeadmin (/tagadmin, /nt)
* [x] /findplayer
* [x] /firstseen (/seenf)
* [x] /seen
* [x] /links (/link, /website, /forum, etc.)
* [x] /list *(Vanilla shadowing)*
* [x] /me *(Vanilla shadowing)*
* [x] /news
* [x] /ping
* [x] /reply (/r)
* [x] /slap
* [x] /msg (/tell, /m, /w) *(Vanilla shadowing)*
* [x] /survival, /creative, and /event
* [x] /togglewelcome (donor)

Features to add:

* [x] Move Player Join/Quit events here (using translatable)
* [x] Server welcome message (Togglable?)

Commands above are subject to expansion.

### PR DESCRIPTION
**Pull request by chuushi**
With #9 done, we may need to make some minor changes to this before we can release this into public.

Additional TODO:
* [x] Make some text more readable (format some text)
* [x] Merge `alert-badge` #14 into here

Finally:
* [ ] Pull Conditions
  * [x] 2 people reviews required
  * [ ] Set `ArcaneBungee v3.0-SNAPSHOT` to `ArcaneBunge v3.0`

Closes #9.

### CODE REVIEW

File: src/main/java/com/arcaneminecraft/bungee/JoinLeaveEvents.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    public void onLoginJoin(PostLoginEvent e) {
+        if (plugin.getSqlDatabase() != null) {
+            plugin.getSqlDatabase().playerJoinThen(e.getPlayer(), (time, oldName) -> {
+                ProxiedPlayer p = e.getPlayer();
+                if (OptionsStorage.get(p, OptionsStorage.Toggles.SHOW_WELCOME_MESSAGE)) {
+                    p.sendMessage(ChatMessageType.SYSTEM, this.welcomeMessage);
+                }
+
+                if (p.hasPermission("arcane.welcome.donor") && OptionsStorage.get(p, OptionsStorage.Toggles.SHOW_DONOR_WELCOME_MESSAGE)) { // TODO: Check permission node existence on ArcaneServer
+                    p.sendMessage(ChatMessageType.SYSTEM, new TextComponent("You are a donor! This message is still a work in progress"));
```
Comment (talrey) [inline src/main/java/com/arcaneminecraft/bungee/JoinLeaveEvents.java]: "This message is still a work in progress" ? Seems fine to me. Maybe mention a donor help command if we've got that as a separate thing?

File: src/main/java/com/arcaneminecraft/bungee/SpyAlert.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+        /* Handy chart:
+         *    sender  |  listener
+         *            | All | Some
+         * -----------+-----+------
+         *        All |  x  |  x
+         *       Some |  x  |  x
+         *     ignore |     |
+         *       None |     |
+         */
```
Comment (talrey) [inline src/main/java/com/arcaneminecraft/bungee/SpyAlert.java:242]: õ_ô
What's the point of _Some_ vs _All_ here???

File: src/main/java/com/arcaneminecraft/bungee/JoinLeaveEvents.java (hunk 1 above)
Comment (chuushi) [reply src/main/java/com/arcaneminecraft/bungee/JoinLeaveEvents.java]: As we discussed in chat, I forgot to port [this](https://github.com/ArcaneMinecraft/ArcaneDonor/blob/master/src/main/java/com/arcaneminecraft/donor/ArcaneDonor.java#L94) over :I

**Review by talrey — COMMENTED** [review]
I mostly skimmed it because it's been a long time and I'm not as familiar with the architecture as I used to be, but it looks good to me.

File: src/main/java/com/arcaneminecraft/bungee/SpyAlert.java (hunk 2 above)
Comment (chuushi) [reply src/main/java/com/arcaneminecraft/bungee/SpyAlert.java:242]: looks like I screwed up on the chart..? ._.

mods will default to some and admins will default to all, and admins can switch to some if needed.

Comment (talrey) [reply src/main/java/com/arcaneminecraft/bungee/SpyAlert.java:242]: So Listener: All should be Admin, and Listener: Some should be Mods with the All row blank?

**Comment by chuushi** [comment]
BungeeCord architecture takes some time to get used to since it's somewhat different from Bukkit/Spigot's architecture. Thanks for the review!

**Pull request merged**